amethyst_assets/src/asset.rs

```diff
@@ -25,7 +27,47 @@ pub trait Asset: Sized {
 }
 
 /// A future for an asset
-pub type AssetFuture<A, E = AssetError> = Shared<Box<Future<Item=A, Error=E>>>;
+pub struct AssetFuture<A>(Shared<Box<Future<Item=A, Error=BoxedErr>>>);
+
+impl<A> Component for AssetFuture<A>
```

Comment:
Do we need this component implementation?

Reply:
Yes. Or you wont be able to attach it to entity.